src/pyhf/constraints.py

```diff
@@ -1,25 +1,41 @@
 from . import get_backend, default_backend
 from . import events
+from .paramview import ParamViewer
 
 
 class gaussian_constraint_combined(object):
-    def __init__(self, pdfconfig):
+    def __init__(self, pdfconfig, batch_size=None):
+        self.batch_size = batch_size
         # iterate over all constraints order doesn't matter....
 
-        self.par_indices = list(range(len(pdfconfig.suggested_init())))
         self.data_indices = list(range(len(pdfconfig.auxdata)))
         self.parset_and_slice = [
             (pdfconfig.param_set(cname), pdfconfig.par_slice(cname))
             for cname in pdfconfig.auxdata_order
         ]
-        self._precompute()
-        events.subscribe('tensorlib_changed')(self._precompute)
 
-    def _precompute(self):
-        tensorlib, _ = get_backend()
+        pnames = [
+            cname
```

Comment:
what is cname here compared to pname? We should label these better...

Reply:
cname is the name of a constrained parameter, pnames is the set of parameters constrained by 'normal'

Reply:
add a comment or rename the variables to be clearer. `pname_constrained` and `pnames_normal` sound better to me -- than `pnames` and `cname`.

Reply:
done